aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/Orchestrator.java

```diff
@@ -180,9 +189,16 @@ private Completable transitionCompletable() {
      */
     public Completable stop() {
         LOG.info("DataStore orchestrator stopping. Current mode = " + currentMode.get().name());
+        if (!startStopSemaphore.tryAcquire()) {
```

Comment:
Given that you generally want calls to stop to be terminal maybe use acquire so calling thread blocks or tryAcquire with timeout for a softer approach?

Reply:
That's a good point. The only place where stop is called in a meaningful way is during a call to DataStore.clear. We'd want to block in that case, so I just went with the acquire option.

Reply:
Thanks again for taking a look. I have refactored this to include a timeout.